### DISCUSSION TITLE
VisualMagick in Visual Studio 2017 (C++)

### DISCUSSION THREAD

**Discussion by yunhyejoung** [ImageMagick/ImageMagick#3366] [Development]
1.  C:\Users\abc\Downloads\ImageMagick-6.8.1\VisualMagick\configure\configure.sln Compiled and built in Visual 2017 (C++).
2. VisualDynamicMT.sln Compiled and built in Visual 2017 (C++).
----> Compile Error 

![error](https://user-images.githubusercontent.com/23716146/110233472-7bc81000-7f67-11eb-9fdc-ccb0e3b528e1.JPG)

----> I have to compile the code below in VS2017.

void CImgmagickDlg::OnBnClickedBtnImgmagick()
{
  const char *path = "C:\\Users\\abc\\Downloads\\ImageMagick-6.8.1\\VisualMagick\\bin";	
  InitializeMagick(path); // CORE_DB_Magick++_.dll 이 있는 경로를 지정하는 것 같다.
  Image worker;
  std::string src = "D:\\sam.bmp";
  worker.read(src);
  worker.resolutionUnits(PixelsPerInchResolution); 
  worker.density("300");
  worker.density(Geometry(3,3));
  worker.write(src);
}**

**Comment by dlemstra** [comment]
This looks like ImageMagick 7 instead of ImageMagick 6. You got the `jpeg-xl` library in the list and that is only available in ImageMagick 7. But that library is not compatible with VS2017 and will require VS2019. We have made sure that in the new source release that when you build with VS2017 that library will no longer be included.

**Reply by yunhyejoung** [reply]
What 8bit, one channel, GrayScale, Image Format in ImageMagick.

Mat CImageMagick::ImageMagickRun()
{
const char *path = "C:\Program Files\ImageMagick-7.0.10-Q16-HDRI";
InitializeMagick(path);

Mat orgImg;
orgImg.create(200, 400, CV_8UC1); // 8bit, 1 channel, Grayscale

// Magick to Mat----------------
Image mgk(orgImg.cols, orgImg.rows, "GRAY", CharPixel, (char *)orgImg.data);
mgk.resolutionUnits(PixelsPerInchResolution);
mgk.density("300");

// Mat to Magick----------------
int w = mgk.columns();
int h = mgk.rows();

Mat opencvImage;
opencvImage.create(h, w, CV_8UC1); // 8bit, 1 channel, Grayscale
// 8Bit, 1Channel Image
mgk.write(0, 0, w, h, "GRAY", Magick::CharPixel, opencvImage.data); //<========== Error
}
